Wrong number of parameters for pageable and sort string queries

[ERROR] org.springframework.data.couchbase.repository.query.StringN1qlQueryCreatorTests.createsQueryCorrectly -- Time elapsed: 0.015 s <<< ERROR!
java.lang.IllegalArgumentException: Number of method parameters (2) must match the number of method invocation arguments (4)
	at org.springframework.data.repository.query.QueryMethodValueEvaluationContextAccessor.collectVariables(QueryMethodValueEvaluationContextAccessor.java:182)
	at org.springframework.data.repository.query.QueryMethodValueEvaluationContextAccessor$DefaultQueryMethodValueEvaluationContextProvider.doGetEvaluationContext(QueryMethodValueEvaluationContextAccessor.java:234)
	at org.springframework.data.repository.query.QueryMethodValueEvaluationContextAccessor$DefaultQueryMethodValueEvaluationContextProvider.getEvaluationContext(QueryMethodValueEvaluationContextAccessor.java:222)
	at org.springframework.data.repository.query.ExtensionAwareQueryMethodEvaluationContextProvider.getEvaluationContext(ExtensionAwareQueryMethodEvaluationContextProvider.java:105)
	at org.springframework.data.couchbase.repository.query.StringBasedN1qlQueryParser.getExpression(StringBasedN1qlQueryParser.java:700)
	at org.springframework.data.couchbase.repository.query.StringBasedN1qlQueryParser.<init>(StringBasedN1qlQueryParser.java:172)
	at org.springframework.data.couchbase.core.query.StringQuery.getStringN1qlQueryParser(StringQuery.java:130)
	at org.springframework.data.couchbase.core.query.StringQuery.toN1qlSelectString(StringQuery.java:75)
	at org.springframework.data.couchbase.repository.query.StringN1qlQueryCreatorTests.createsQueryCorrectly(StringN1qlQueryCreatorTests.java:120)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

--- a/src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java
+++ b/src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java
@@ -711,8 +711,11 @@ public class StringBasedN1qlQueryParser {
 		for (Object o : accessor) {
 			params.add(o);
 		}
-		params.add(accessor.getPageable());
-		params.add(accessor.getSort());
+		if( accessor.getPageable().isPaged()) {
+			params.add(accessor.getPageable());
+		} else if (accessor.getSort().isSorted()) {
+			params.add(accessor.getSort());
+		}
 		return params.toArray();
 	}
 }

--- a/src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java
+++ b/src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java
@@ -40,6 +40,7 @@ import org.springframework.data.couchbase.repository.Scope;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.domain.Slice;
+import org.springframework.data.domain.Sort;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
@@ -153,6 +154,11 @@ public interface AirportRepository extends CouchbaseRepository<Airport, String>,
 	@ScanConsistency(query = QueryScanConsistency.REQUEST_PLUS)
 	@Query("#{#n1ql.selectEntity} WHERE #{#n1ql.filter} AND iata != $1")
 	Page<Airport> getAllByIataNot(String iata, Pageable pageable);
+
+	@ScanConsistency(query = QueryScanConsistency.REQUEST_PLUS)
+	@Query("#{#n1ql.selectEntity} WHERE #{#n1ql.filter} AND iata != $1")
+	List<Airport> getAllByIataNotSort(String iata, Sort sort);
+
 
 	@ScanConsistency(query = QueryScanConsistency.REQUEST_PLUS)
 	@Query("SELECT iata, \"\" as __id, 0 as __cas from #{#n1ql.bucket} WHERE #{#n1ql.filter} order by meta().id")

--- a/src/test/java/org/springframework/data/couchbase/repository/CouchbaseRepositoryQueryIntegrationTests.java
+++ b/src/test/java/org/springframework/data/couchbase/repository/CouchbaseRepositoryQueryIntegrationTests.java
@@ -885,6 +885,31 @@ public class CouchbaseRepositoryQueryIntegrationTests extends ClusterAwareIntegr
 	}
 
 	@Test
+	void countSlicePageTest() {
+		airportRepository.withOptions(QueryOptions.queryOptions().scanConsistency(REQUEST_PLUS)).deleteAll();
+		String[] iatas = { "JFK", "IAD", "SFO", "SJC", "SEA", "LAX", "PHX" };
+
+		airportRepository.countOne();
+		try {
+			airportRepository.saveAll(
+				Arrays.stream(iatas).map((iata) -> new Airport("airports::" + iata, iata, iata.toLowerCase(Locale.ROOT)))
+					.collect(Collectors.toSet()));
+
+		Pageable sPageable = PageRequest.of(0, 2).withSort(Sort.by("iata"));
+		Page<Airport> sPage = airportRepository.getAllByIataNot("JFK", sPageable);
+		System.out.println(sPage);
+
+		Sort sort = Sort.by("iata");
+		List<Airport> sList = airportRepository.getAllByIataNotSort("JFK", sort);
+		System.out.println(sList);
+
+		} finally {
+			airportRepository
+				.deleteAllById(Arrays.stream(iatas).map((iata) -> "airports::" + iata).collect(Collectors.toSet()));
+		}
+	}
+
+	@Test
 	void countSlicePage() {
 		airportRepository.withOptions(QueryOptions.queryOptions().scanConsistency(REQUEST_PLUS)).deleteAll();
 		String[] iatas = { "JFK", "IAD", "SFO", "SJC", "SEA", "LAX", "PHX" };

